webnn/resources/test_data/l2_pool2d.json

```diff
@@ -0,0 +1,1509 @@
+            84.59886932373047,
+            12.081207275390625,
+            37.8386344909668,
+            57.85399627685547,
+            28.702816009521484,
+            74.17911529541016,
+            67.8487319946289,
+            0.811935305595398,
+            79.18359375,
+            83.0001449584961,
+            82.172119140625,
+            92.50651550292969,
+            8.42988395690918,
+            93.89906311035156,
+            58.69465255737305
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "windowDimensions": [3, 3],
+        "strides": [2, 2],
+        "autoPad": "same-lower"
+      },
+      "expected": {
+        "name": "output",
+        "shape": [1, 2, 2, 2],
+        "data": [
+          127.90882873535156,
+          185.0220947265625,
+          105.7017593383789,
+          164.85511779785156,
+          135.8782501220703,
+          140.2302703857422,
+          138.09474182128906,
+          207.78431701660156
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "l2Pool2d float32 4D tensor options.autoPad=same-upper ignores options.padding",
+      "inputs": {
+        "input": {
+          "shape": [1, 2, 4, 4],
+          "data": [
+            89.69132995605469,
+            49.80897521972656,
+            84.17794036865234,
+            98.6054458618164,
+            25.800601959228516,
+            71.89942932128906,
+            95.26441192626953,
+            26.428104400634766,
+            44.16105270385742,
+            24.04642105102539,
+            30.115402221679688,
+            54.41345977783203,
+            47.779483795166016,
+            22.942420959472656,
+            24.280946731567383,
+            81.68242645263672,
+            84.46887969970703,
+            84.59886932373047,
+            12.081207275390625,
+            37.8386344909668,
+            57.85399627685547,
+            28.702816009521484,
+            74.17911529541016,
+            67.8487319946289,
+            0.811935305595398,
+            79.18359375,
+            83.0001449584961,
+            82.172119140625,
+            92.50651550292969,
+            8.42988395690918,
+            93.89906311035156,
+            58.69465255737305
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "windowDimensions": [3, 3],
+        "padding": [2, 2, 2, 2],
+        "strides": [2, 2],
+        "autoPad": "same-upper"
+      },
+      "expected": {
+        "name": "output",
+        "shape": [1, 2, 2, 2],
+        "data": [
+          189.6240234375,
+          174.50064086914062,
+          82.66900634765625,
+          105.49571228027344,
+          193.0572967529297,
+          159.13827514648438,
+          174.94363403320312,
+          160.9452667236328
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "l2Pool2d float32 4D tensor options.autoPad=same-lower ignores options.padding",
```

Comment:
There was talk of removing automagic padding consistently for both convolution and pooling: https://github.com/webmachinelearning/webnn-baseline/pull/68#pullrequestreview-1817721795
I suppose we can keep these cases until a spec CR is proposed.